src/ORM/Association.php

```diff
@@ -749,6 +749,15 @@ public function attachTo(Query $query, array $options = []): void
                     $this->getName()
                 ));
             }
+            if (
+                !empty($options['matching']) &&
+                $this->_strategy === static::STRATEGY_JOIN &&
+                $dummy->getContain()
+            ) {
+                throw new RuntimeException(
+                    "Cannot contain associations in query returned by query builder for `{$this->getName()}`."
```

Comment:
```suggestion
                    "Cannot contain associations in queries returned by query builder Closures for `{$this->getName()}`."
```

Reply:
Should we fix the other exception messages to use `Closure`?

`'Query builder for association "%s" did not return a query'`

Reply:
Never mind -- closure isn't really relevant. I adjusted the test to happen no matter how the contain is set.